src/ai/defaultai_seafaring.cc

```diff
@@ -227,21 +225,38 @@ bool DefaultAI::marine_main_decisions(const Time& gametime) {
 		}
 	}
 
+	if (game().naval_warfare_allowed() && ports_finished_count * 2 > warships_count &&
+	    tradeships_count > expeditions_in_progress + ports_count / 3 + 1) {
+		warship_needed = true;
+	}
+	if (!ship_free && warships_count > 0) {
+		tradeship_refit_needed = true;
+	}
+
 	// starting an expedition? if yes, find a port and order it to start an expedition
+	bool start_expedition = false;
 	if (ports_count > 0 && expeditions_in_progress == 0 && expeditions_in_prep == 0 &&
 	    !persistent_data->no_more_expeditions && ship_free && basic_economy_established) {
```

Comment:
correct but I believe that is currently out of scope counting my time and the upcoming release. So as for diplomacy in the first release when it was implemented this is just a very rudimentary solution.      
I am not much in favor to do much more then getting the ports garrison done as we still might face big changes in naval warfare.

Reply:
I think this will already go a long way while also allowing escape from the first island where needed:

```suggestion
	    !persistent_data->no_more_expeditions && (ports_count < 2 ? ship_free : !need_ship) && basic_economy_established) {
```

Reply:
good idea